src/core/mount.c

```diff
@@ -899,10 +899,10 @@ static int mount_spawn(Mount *m, ExecCommand *c, PidRef *ret_pid) {
         return 0;
 }
 
-static void mount_enter_dead(Mount *m, MountResult f) {
+static void mount_enter_dead(Mount *m, MountResult f, bool flush_failure) {
```

Comment:
`flush_failure` seems like a bad name since this will flush "success" as well. I would use `flush_result` instead.